### PR TITLE
feat!: auto-centered ToolTips

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Changes the tool tips to always have their beaks horizontally centered on their targets.

This way we won't have to annoyingly & imprecisely manually specify horizontal offsets in styles.

Also renames the positioning prop from `position`, which conflicts with CSS-in-JS, to `alignment`.

Corresponding monolith PR: [Used auto-centering ToolTip from Gamut](https://github.com/codecademy-engineering/Codecademy/pull/23635).

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/k68iQ6kmG5fLqSY1kTVCyn/Tooltip?node-id=2%3A38
- [x] Related to JIRA ticket: WEB-1389
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.9-alpha.73c584.0<br/>@codecademy/gamut-labs@13.4.1-alpha.73c584.0<br/>@codecademy/gamut@29.10.0-alpha.73c584.0<br/>@codecademy/styleguide@33.10.0-alpha.73c584.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070cdc4f0ca0d00ac6c4071--gamut-preview.netlify.app>https://6070cdc4f0ca0d00ac6c4071--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070cdc4f0ca0d00ac6c4071>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]

File: packages/styleguide/stories/Atoms/ToolTip.stories.mdx [hunk 1]
```diff
@@ -44,21 +45,25 @@ Great for surfacing additional information that may or may not be relevant _with
...
-            <TooltipArea key={position} tipPosition={position}>
-              {position.split('-').map(upperFirst).join(' ')}: &nbsp;
-              <ToolTip {...args} position={position} />
+            <TooltipArea key={alignment} tipAlignment={alignment}>
+              {alignment.split('-').map(upperFirst).join(' ')}: &nbsp;
+              <ToolTip {...args} alignment={alignment} />
+              <ToolTip
+                {...args}
+                alignment={alignment}
+                target={<>Hi: {args.target}</>}
```
Comment (dreamwasp) [inline packages/styleguide/stories/Atoms/ToolTip.stories.mdx:65]: these examples are a bit hard to read right next to each other, maybe just make this text a little longer so the difference is more noticeable

**Review by dreamwasp — APPROVED** [review]
✅

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.16-alpha.0fc57f.0<br/>@codecademy/gamut-labs@13.4.8-alpha.0fc57f.0<br/>@codecademy/gamut@29.10.0-alpha.0fc57f.0<br/>@codecademy/styleguide@33.10.0-alpha.0fc57f.0<br/></pre>

File: packages/styleguide/stories/Atoms/ToolTip.stories.mdx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Atoms/ToolTip.stories.mdx:65]: 👌 

More padding left before the Hi!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.16-alpha.4f30ec.0<br/>@codecademy/gamut-labs@13.4.8-alpha.4f30ec.0<br/>@codecademy/gamut@29.10.0-alpha.4f30ec.0<br/>@codecademy/styleguide@33.10.0-alpha.4f30ec.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60761d3e920fae008fa72c98--gamut-preview.netlify.app>https://60761d3e920fae008fa72c98--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60761d3e920fae008fa72c98>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.16-alpha.4a0b38.0<br/>@codecademy/gamut-labs@13.4.8-alpha.4a0b38.0<br/>@codecademy/gamut@29.10.0-alpha.4a0b38.0<br/>@codecademy/styleguide@33.10.0-alpha.4a0b38.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6077004cddcba256a0131a20--gamut-preview.netlify.app>https://6077004cddcba256a0131a20--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6077004cddcba256a0131a20>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.26-alpha.3543ac.0<br/>@codecademy/gamut-labs@13.5.3-alpha.3543ac.0<br/>@codecademy/gamut@30.2.0-alpha.3543ac.0<br/>@codecademy/styleguide@34.3.0-alpha.3543ac.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60785ba2d7bcf031cd120c31--gamut-preview.netlify.app>https://60785ba2d7bcf031cd120c31--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60785ba2d7bcf031cd120c31>Deploy Logs</a></p>

**Pull request merged**